src/models/plugin-state.ts

```diff
@@ -1,28 +1,63 @@
-import { getAllItems, IResult } from "@concord-consortium/codap-plugin-api";
-import { makeAutoObservable } from "mobx";
+import { getAllItems, getCaseBySearch, IResult } from "@concord-consortium/codap-plugin-api";
+import { makeAutoObservable, reaction } from "mobx";
 import {
+  kDataContextName,
+  kMapPinsCollectionName,
   kPinColorAttributeName, kPinDataContextName, kPinLatAttributeName, kPinLongAttributeName
 } from "../data/constants";
+import { createSelectionList, deleteSelectionList, updateSelectionList } from "../utils/codap-utils";
```

Comment:
This usage of the codap-utils by the plugin state is causing a dependency cycle. 
I'm not sure what the best fix is.

One option I can think of is to move the `initializeNeoPlugin` function from codap-utils into plugin-state. I think this will remove any dependency by codap-utils on plugin-state. And then it will be fine for plugin-state to depend on codap-utils.